### ISSUE TITLE
The field 'indexes' must be an array, but got object

### ISSUE THREAD

**Issue by timarandras** [deepstreamIO/deepstream.io#970]
Trying to use V4 server and client with @deepstream/storage-mongodb@v4 and got this error:
```
(node:9297) DeprecationWarning: current URL string parser is deprecated, and will be removed in a future version. To use the new parser, pass option { useNewUrlParser: true } to MongoClient.connect.
{ MongoError: The field 'indexes' must be an array, but got object
    at Connection.<anonymous> (/home/timar/prog/moc/ds-server/node_modules/mongodb-core/lib/connection/pool.js:443:61)
    at Connection.emit (events.js:198:13)
    at processMessage (/home/timar/prog/moc/ds-server/node_modules/mongodb-core/lib/connection/connection.js:364:10)
    at Socket.<anonymous> (/home/timar/prog/moc/ds-server/node_modules/mongodb-core/lib/connection/connection.js:533:15)
    at Socket.emit (events.js:198:13)
    at addChunk (_stream_readable.js:288:12)
    at readableAddChunk (_stream_readable.js:269:11)
    at Socket.Readable.push (_stream_readable.js:224:10)
    at TCP.onStreamRead [as onread] (internal/stream_base_commons.js:94:17)
  ok: 0,
  errmsg: 'The field \'indexes\' must be an array, but got object',
  code: 14,
  codeName: 'TypeMismatch',
  name: 'MongoError',
  [Symbol(mongoErrorContextSymbol)]: {} }
TypeError: Cannot set property '__ds' of undefined
    at Object.module.exports.transformValueForStorage (/home/timar/prog/moc/ds-server/node_modules/@deepstream/storage-mongodb/src/transform-data.js:28:15)
    at DSStorage.Connector.set (/home/timar/prog/moc/ds-server/node_modules/@deepstream/storage-mongodb/src/connector.js:110:25)
    at DSStorage.storage.set (/home/timar/prog/moc/ds-server/node_modules/@deepstream/server/src/config/config-initialiser.ts:362:12)
    at RecordHandler.create (/home/timar/prog/moc/ds-server/node_modules/@deepstream/server/src/handlers/record/record-handler.ts:425:29)
    at RecordHandler.onPermissionResponse (/home/timar/prog/moc/ds-server/node_modules/@deepstream/server/src/handlers/record/record-handler.ts:645:7)
    at RuleApplication.run (/home/timar/prog/moc/ds-server/node_modules/@deepstream/server/src/services/permission/valve/rule-application.ts:102:19)
    at new RuleApplication (/home/timar/prog/moc/ds-server/node_modules/@deepstream/server/src/services/permission/valve/rule-application.ts:64:10)
    at DSPermission.canPerformAction (/home/timar/prog/moc/ds-server/node_modules/@deepstream/server/src/services/permission/valve/config-permission.ts:142:5)
    at RecordHandler.permissionAction (/home/timar/prog/moc/ds-server/node_modules/@deepstream/server/src/handlers/record/record-handler.ts:605:30)
    at RecordHandler.onCreateOrReadComplete (/home/timar/prog/moc/ds-server/node_modules/@deepstream/server/src/handlers/record/record-handler.ts:277:12)
```
Deepstream server setup using mongo connector:
```javascript
const MongoDBStorageConnector = require('@deepstream/storage-mongodb');

const server = new Deepstream({/*...*/});

server.set('storage', new MongoDBStorageConnector({
        connectionString: "mongodb://username:s3cr3t@mongo.example.com:27017/ds-records?retryWrites=true&authSource=ds-records",
        splitChar: '/',
      }));

server.start();
```

@deepstream/server: ^4.0.0-rc.17
@deepstream/client: ^4.0.0-rc.42
@deepstream/storage-mongodb: ^2.0.1
@deepstream/cache-redis: ^2.0.0-rc.12

node -v
v10.16.0

**Comment by timarandras** [comment]
Just made a demo in Codesandbox using Mongo Atlas free tier:

```javascript
const MongoStorageConnector = require('@deepstream/storage-mongodb');
const { Deepstream } = require('@deepstream/server');
const deepstreamClient = require('@deepstream/client');
const server = new Deepstream({
  connectionEndpoints: [
    {
      type: 'uws-websocket',
      options: {
        headers: ['cookie', 'user-agent', 'host', 'origin'],
      },
    },
  ],
});

server.set(
  'storage',
  new MongoStorageConnector({
    connectionString:
      'mongodb+srv://test_user974:bmQSgQQk87SqYyZH@cluster0-yix1k.mongodb.net/test?retryWrites=true&w=majority',
    splitChar: '/',
  }),
);

server.on('started', async () => {
  console.log('DS Server started');

  const client = deepstreamClient('localhost:6020');
  client.login();

  try {
    const record = client.record.getRecord('myRecord/apple');
    //    await record.whenReady();
  } catch (error) {
    console.error(error);
  }
});

server.start();
```
[Open in Codesandbox](https://codesandbox.io/embed/deepstream-mongodb-error-server-bkmem)

**Comment by yasserf** [comment]
fixed in v2.0.2, thanks for demo!
